### ISSUE TITLE
`T_and_F_symbol_linter()` false positive with formula syntax

### ISSUE THREAD

**Issue by IndrajeetPatil** [r-lib/lintr#2637]
``` r
library(lintr)
lint(
  text = "ivreg::ivreg(Q ~ P + D | D + F + A, data = Kmenta)", 
  linters = T_and_F_symbol_linter()
)
#> <text>:1:31: style: [T_and_F_symbol_linter] Use FALSE instead of the symbol F.
#> ivreg::ivreg(Q ~ P + D | D + F + A, data = Kmenta)
#>                              ~^
```

The `F` here corresponds to the column name:

```r
data(Kmenta, package = "ivreg")
colnames(Kmenta)
#> [1] "Q" "P" "D" "F" "A"
```

<sup>Created on 2024-07-28 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

**Comment by MichaelChirico** [comment]
should be safe to assume T/F are not TRUE/FALSE in the _top-level_ of a formula, but  not in calls like y~foo(x, arg=TRUE)
